data/functions.sql

```diff
@@ -292,3 +293,77 @@ BEGIN
   RETURN new_tags;
 END;
 $$ LANGUAGE plpgsql IMMUTABLE;
+
+-- returns the (possibly fractional) zoom at which the given way
+-- area will be one square pixel nominally on screen (assuming
+-- that tiles are 256x256px at integer zooms). sadly, features
+-- aren't always rectangular and axis-aligned, but this should
+-- still give a reasonable approximation to the zoom that it
+-- would be appropriate to show them.
+CREATE OR REPLACE FUNCTION mz_one_pixel_zoom(
+  way_area real)
+RETURNS real AS $$
+BEGIN
+  RETURN (17.256-ln(way_area)/ln(4));
+END;
+$$ LANGUAGE plpgsql IMMUTABLE;
+
+-- calculate the collapse of several properties onto one string
+-- 'kind'. this involves a series of precedence choices, as
+-- it's possible for a feature to have values in several of
+-- these categories. in general, the "most important" should go
+-- first, or its presence should modify the later.
+CREATE OR REPLACE FUNCTION mz_calculate_landuse_kind(
+  landuse_val text,
+  leisure_val text,
+  natural_val text,
+  highway_val text,
+  aeroway_val text,
+  amenity_val text,
+  tourism_val text,
+  man_made_val text,
+  power_val text,
+  boundary_val text)
+RETURNS text AS $$
+BEGIN
+  RETURN
+    CASE
+      WHEN boundary_val IN (
+        'national_park', 'protected_area')
+	THEN boundary_val
+      WHEN landuse_val IN (
+        'park', 'forest', 'residential', 'retail', 'commercial', 'industrial',
+	'railway', 'cemetery', 'grass', 'farmyard', 'farm', 'farmland', 'wood',
+	'meadow', 'village_green', 'recreation_ground', 'allotments', 'quarry',
+	'urban', 'rural', 'military')
+        THEN landuse_val
+      WHEN leisure_val IN (
+        'park', 'garden', 'playground', 'golf_course', 'sports_centre', 'pitch',
+	'stadium', 'common', 'nature_reserve')
+	THEN leisure_val
+      WHEN natural_val IN (
+        'wood', 'land', 'scrub', 'wetland', 'glacier')
+	THEN natural_val
+      WHEN highway_val IN (
```

Comment:
Because `highway` was previously included in the `COALESCE` (e.g: [here](https://github.com/mapzen/vector-datasource/pull/227/files#diff-21e9ac7608f051b94d3add540d651c99L44)), and is selected as part of [mz_calculate_is_landuse](https://github.com/mapzen/vector-datasource/pull/227/files#diff-85235d701f80586d926a4211261dbbc1L16).